File: Laboratory_2/Swift/task3.swift
import Foundation

func isNumber(a: Int) -> Bool {
    /* Проверяет, что это число, а не цифра. */
    return a / 10 != 0
}

func cntOddDigits(a: Int) -> Int {
    /* Считает количество нечетных цифр в числе. */
    var res = 0
    var number = abs(a)

    while number > 0 {
        res += number % 2
        number /= 10
    }
    return res
}

// Основная часть программы
print("Введите количество чисел: ", terminator: "")
if let input = readLine(), let n = Int(input) {
    for i in 0..<n {
        print("Введите \(i + 1) число: ", terminator: "")
        if let numInput = readLine(), let num = Int(numInput) {
            
            var cntOdd = 0
            if isNumber(a: num) {
                cntOdd = cntOddDigits(a: num)
            } else {
                print("\(num) - это цифра!")
            }

            print("Количество нечетных цифр: \(cntOdd)")
        }
    }
}

File: LaboratoryWork/Laboratory_2/Python/task3.py
def is_number(a: int) -> bool:
    '''Проверяет, что это число, а не цифра.'''
    return a // 10 != 0


def cnt_odd_digits(a: int) -> int:
    '''Считает количество нечетных цифр в числе'''
    res = 0
    a = abs(a)

    while a > 0:
        res += (a % 2)
        a //= 10
    return res


def main():
    while True:  # проверка на дурака
        try:
            n = int(input("Введите количество чисел: "))
            if n > 0:
                break
            print("N - положительное число!")
        except ValueError:
            print("Ваш ввод некорректен, введите целое число!")
    
    for _ in range(n):
        while True:
            try:
                num = int(input("Введите число: "))
                break
            except ValueError:
                print("Введите целое число!")

        cnt_odd = 0
        if is_number(num):
            cnt_odd = cnt_odd_digits(num)
        else:
            print(f"{num} - это цифра!")
        print(f"Количество нечетных цифр: {cnt_odd}")


if __name__ == "__main__":
    main()